server/server.go

```diff
@@ -121,8 +130,10 @@ func UserRegister(w http.ResponseWriter, r *http.Request) {
 	}
 	fmt.Println(registerRequest)
 	u := model.User{}
+
+	hash, _ := HashPassword(registerRequest.Password)
 	u.Username = registerRequest.Username
-	u.Password = registerRequest.Password
+	u.Password = hash
```

Comment:
Having a separate api model `UserRegisterRequest` and database model `User` is useful here. We can have a `Password` field in `UserRegisterRequest` and change `User` to have a `PasswordHash` field to relay intent.